### DISCUSSION TITLE
Sidebars

### DISCUSSION THREAD

**Discussion by schalkneethling** [mdn/yari#4033] [Ideas]
Sidebars on MDN Web Docs have long been a thorn in the side both from a visual and UX perspective and from an author and developer perspective. We can do some small things to improve the current situation, but it really needs a rethink.

~~We are currently embarking on a redesign of MDN Web Docs and as part of this, we would love to make sidebars better in as many areas as possible. We would love to get your thoughts, feedback, and suggestions.~~

Some existing files issues:

- https://github.com/mdn/yari/issues/1721
- https://github.com/mdn/yari/issues/2221

**Comment by schalkneethling** [comment]
A comment from a reader: https://github.com/mdn/yari/issues/2221#issuecomment-861304555

**Reply by wbamberg** [reply]
About wrapping of entries in the JS sidebar. Rather than:

```
String
    Methods
        String.prototype[@@iterator]()
        String.prototype.at()
        String.prototype.charAt()
...
        String.fromCharCode()
        String.fromCodePoint()
        String.prototype.includes()
```

...would it not be better to have:

```
String
    Static methods
        fromCharCode()
        fromCodePoint()
...
    Instance methods
        [@@iterator]()
        at()
        charAt()
        includes()
```

**Reply by its4zahoor** [reply]
Yeah! I had proposed a similar idea in the [#2221 Comment](https://github.com/mdn/yari/issues/2221#issuecomment-861304555). This is a common practice in many websites to show the method name (e.g includes) only in the title/sidebar and then use the full name (e.g String.prototype.includes) in the code examples.

Additionally, a sidebar with collapsible sections will be a great UX. Where sections could be Methods, Properties, Instance Methods, Iterators etc

**Reply by teoli2003** [reply]
This kind of sidebars could work on api/ too (even if we don't have the .prototype, the wrapping still happens sometimes).

Also we can list inherited methods/properties in a transparent way: the devs doesn't really care if it is inherited or not, in the list.

**Comment by schalkneethling** [comment]
Hey @wbamberg, @its4zahoor, and @teoli2003. I started a [new discussion](https://github.com/mdn/mdn-community/discussions/69) on the new [mdn-community](https://github.com/mdn/mdn-community) repo to further discuss this topic and reach an actionable conclusion. Your participation there will be appreciated.

**Reply by wbamberg** [reply]
@schalkneethling that discussion seems to be about icons, not sidebars?
